packages/elements/src/components/ino-markdown-editor/markdown-serializer.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  defaultMarkdownParser,
+  defaultMarkdownSerializer,
+  MarkdownParser,
+  MarkdownSerializer,
+  TokenConfig,
+} from 'prosemirror-markdown';
+import { Node as ProsemirrorNode, Schema } from 'prosemirror-model';
+import Bold from '@tiptap/extension-bold';
+import Code from '@tiptap/extension-code';
+import Italic from '@tiptap/extension-italic';
+import Strike from '@tiptap/extension-strike';
+import Blockquote from '@tiptap/extension-blockquote';
+import Heading from '@tiptap/extension-heading';
+import HorizontalRule from '@tiptap/extension-horizontal-rule';
+import Paragraph from '@tiptap/extension-paragraph';
+import OrderedList from '@tiptap/extension-ordered-list';
+import BulletList from '@tiptap/extension-bullet-list';
+import ListItem from '@tiptap/extension-list-item';
+import CodeBlock from '@tiptap/extension-code-block';
+import HardBreak from '@tiptap/extension-hard-break';
+import Link from '@tiptap/extension-link';
+
+// Source taken from
+// https://gitlab.com/gitlab-org/gitlab/-/blob/master/app/assets/javascripts/content_editor/services/markdown_serializer.js
+const defaultSerializerConfig = {
+  marks: {
+    em: defaultMarkdownSerializer.marks.em,
+    [Bold.name]: defaultMarkdownSerializer.marks.strong,
+    [Code.name]: defaultMarkdownSerializer.marks.code,
+    [Italic.name]: {
+      open: '_',
+      close: '_',
+      mixable: true,
+      expelEnclosingWhitespace: true,
+    },
+    [Link.name]: defaultMarkdownSerializer.marks.link,
+    [Strike.name]: {
+      open: '~~',
+      close: '~~',
+      mixable: true,
+      expelEnclosingWhitespace: true,
+    },
+  },
+  nodes: {
+    ...defaultMarkdownSerializer.nodes,
+    [Blockquote.name]: (state, node) => {
+      if (node.attrs.multiline) {
+        state.write('>>>');
+        state.ensureNewLine();
+        state.renderContent(node);
+        state.ensureNewLine();
+        state.write('>>>');
+        state.closeBlock(node);
+      } else {
+        state.wrapBlock('> ', null, node, () => state.renderContent(node));
+      }
+    },
+    [CodeBlock.name]: defaultMarkdownSerializer.nodes.code_block,
+    [Heading.name]: defaultMarkdownSerializer.nodes.heading,
+    [HorizontalRule.name]: defaultMarkdownSerializer.nodes.horizontal_rule,
+    [BulletList.name]: defaultMarkdownSerializer.nodes.bullet_list,
+    [OrderedList.name]: defaultMarkdownSerializer.nodes.ordered_list,
+    [ListItem.name]: defaultMarkdownSerializer.nodes.list_item,
+    [Paragraph.name]: defaultMarkdownSerializer.nodes.paragraph,
+    [Text.name]: defaultMarkdownSerializer.nodes.text,
```

Comment:
Using `Shift + Enter` in Preview-Mode for hard line breaks leads to the following error: `Uncaught Error: Token type 'hardBreak' not supported by Markdown renderer` and the preview text cannot be parsed to markdown code.
As a user, I would expect it to either work or print a message that this action is not supported.

Reply:
I fix that by adding the `HardBreak`. Beside this I add an error handling if parsing or serializing throws an error. The error message occurs by a simple alter box